README.md

```diff
@@ -1113,6 +1113,37 @@ In either case:
        an attribute when `self` is an ActiveRecord model: `self.guest = user`.
     3. Referencing the current instance's class: `self.class`.
 
+* <a name="freeze-constants"></a>When defining arrays and hashes meant to be
+    constants, make sure to call `freeze` on them. ([More on this][ruby-freeze].)
+    <sup>[[link](#freeze-constants)]</sup>
+
+    The reason is that Ruby constants are actually mutable. Calling `freeze`
+    ensures they are not mutated and are therefore truly constant and 
+    attempting to modify them will raise an exception. 
+
+    ```ruby
+    class COLOR
+      # Each color
+      RED = 0
```

Comment:
some types are already frozen by default, like numbers and symbols.
```
2.4.0 :001 > 0.frozen?
 => true
2.4.0 :002 > :test.frozen?
 => true
```
and freezing doesn't actually prevent reassignment. i'm not sure if there is any way to prevent reassignment at all, short of choosing a paradigm, I think
```
2.4.0 :003 > x = 'a'.freeze
 => "a"
2.4.0 :004 > x = 'b'.freeze
 => "b"
```